app/src/main/cpp/skyline/services/am/storage/IStorage.h

```diff
@@ -1,23 +1,34 @@
+
 // SPDX-License-Identifier: MPL-2.0
 // Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
 
 #pragma once
 
 #include <services/serviceman.h>
+#include <common/span.h>
 
 namespace skyline::service::am {
     /**
      * @brief IStorage is used to open an IStorageAccessor to access a region of memory
      * @url https://switchbrew.org/wiki/Applet_Manager_services#IStorage
      */
-    class IStorage : public BaseService, public std::enable_shared_from_this<IStorage> {
+    class IStorage : public BaseService {
       private:
         size_t offset{}; //!< The current offset within the content for pushing data
 
+      protected:
+        IStorage(const DeviceState &state, ServiceManager &manager, bool writable);
+
       public:
-        std::vector<u8> content; //!< The container for this IStorage's contents
 
-        IStorage(const DeviceState &state, ServiceManager &manager, size_t size);
+        virtual ~IStorage();
+
+        /**
+         * @brief All accesses happen through this
+         */
```

Comment:
I feel the comment isn't really functionally stating what this does but is rather a meta commentary on the use of it, `@return A span of the backing storage for this IStorage` would be more apt.